client/keys/show.go

```diff
@@ -18,30 +22,61 @@ const (
 	FlagPublicKey = "pubkey"
 	// FlagBechPrefix defines a desired Bech32 prefix encoding for a key.
 	FlagBechPrefix = "bech"
+
+	flagMultiSigThreshold = "multisig-threshold"
 )
 
+var _ keys.Info = (keys.Info)(nil)
+
+type multiSigKey struct {
+	name string
+	key  crypto.PubKey
+}
+
+func (m multiSigKey) GetName() string            { return m.name }
+func (m multiSigKey) GetType() keys.KeyType      { return keys.TypeLocal }
+func (m multiSigKey) GetPubKey() crypto.PubKey   { return m.key }
+func (m multiSigKey) GetAddress() sdk.AccAddress { return sdk.AccAddress(m.key.Address()) }
+
 func showKeysCmd() *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "show [name]",
 		Short: "Show key info for the given name",
 		Long:  `Return public details of one local key.`,
-		Args:  cobra.ExactArgs(1),
+		Args:  cobra.MinimumNArgs(1),
 		RunE:  runShowCmd,
 	}
 
 	cmd.Flags().String(FlagBechPrefix, "acc", "The Bech32 prefix encoding for a key (acc|val|cons)")
 	cmd.Flags().Bool(FlagAddress, false, "output the address only (overrides --output)")
 	cmd.Flags().Bool(FlagPublicKey, false, "output the public key only (overrides --output)")
+	cmd.Flags().UintP(flagMultiSigThreshold, "m", 1, "K out of N required signatures")
 
 	return cmd
 }
 
-func runShowCmd(cmd *cobra.Command, args []string) error {
-	name := args[0]
+func runShowCmd(cmd *cobra.Command, args []string) (err error) {
+	var info keys.Info
 
-	info, err := GetKeyInfo(name)
-	if err != nil {
-		return err
+	if len(args) == 1 {
+		info, err = GetKeyInfo(args[0])
+		if err != nil {
+			return err
+		}
+	} else {
+		pks := make([]crypto.PubKey, len(args))
```

Comment:
shouldn't we check that the flag value for `multisig-threshold == len(args)` ?

Reply:
otherwise return an error

Reply:
it should actually be `0 < K <= len(args)`. I'll do some more testing :+1:

Reply:
Done